packages/admin/cms-admin/src/generator/future/generateFormField.ts

```diff
@@ -60,14 +66,15 @@ export function generateFormField(
             {(props) => (
                 <FormControlLabel
                     label={<FormattedMessage id="${instanceGqlType}.${name}" defaultMessage="${label}" />}
-                    control={<FinalFormCheckbox {...props} />}
+                    control={<FinalFormCheckbox ${config.readOnly ? readOnlyProps : ""} {...props} />}
```

Comment:
It's kinda odd that FinalFormCheckbox doesn't use readOnly and disabled from Field, but all other components do 🤔

Reply:
Looks like we have a bigger issue here, see: COM-503